### ISSUE TITLE
Instant Recognition gives wrong label, but testing directly through Caffe gives accuracy = 1 for same image

### ISSUE THREAD

**Issue by sbanerj1** [BVLC/caffe#2773]
Hi,

I have been running Caffe to recognize images related to pharmaceutical drugs. I train the Caffenet architecture with my own data of 260 training images for 13 classes (20 images/class). I have saved the model state at the 10,000th iteration to do instant and batch recognition with it. 

For a test set of 10 images (all of same drug), I make a leveldb folder using the create_imagenet.sh file with the label file for those 10 images. I test it with the saved model (10000th itr) and 9 of them give accuracy of 1. The results can be seen below:-

![batch1](https://cloud.githubusercontent.com/assets/6412439/8703480/4ab32b16-2aeb-11e5-8f8f-d01b08899203.jpg)

But when I try to do instant recognition with each image at a time, the labels it churns out is wrong. Here's the code I use for that:-

![code1](https://cloud.githubusercontent.com/assets/6412439/8703521/8c6fb790-2aeb-11e5-9b0a-044e17bb108e.jpg)

Could you kindly let me know what am I doing wrong? Do I need to change the deploy.prototxt file before I do instant recognition?

Thanks in advance.

**Comment by smartbitcoin** [comment]
It's better to use caffe user group to asking for help.  Git issue is for something like bug report or new feature request.

it's better you post your model here.  what's your layer "prob"?  why you call out['prob'].argmax()? if your output layer is multi-to-one,  the output should be directly label like [1...10] etc.

**Comment by sbanerj1** [comment]
@smartbitcoin Thanks for replying. I realized that this was a wrong place to post the issue only after actually posting it, I have shared it with Caffe users Google group as well.

I wanted to get probability scores for each test image, so I followed instructions as posted here: https://github.com/BVLC/caffe/blob/master/examples/00-classification.ipynb.

I use the standard 'Caffenet' model shared with Caffe for training with my data. Here's the train_val.prototxt which I use for training as can be found here: https://github.com/BVLC/caffe/blob/master/models/bvlc_reference_caffenet/train_val.prototxt
I have only changed the top part to take in my data for training and testing, as shown below:-
name: "CaffeNet"
layers {
  name: "data"
  type: DATA
  top: "data"
  top: "label"
  data_param {
    source: "Sandipan1_half_train20_leveldb"
    backend: LMDB
    batch_size: 260
  }
  transform_param {
    crop_size: 227
    mean_file: "imagenet_mean.binaryproto"
    mirror: true
  }
  include: { phase: TRAIN }
}
layers {
  name: "data"
  type: DATA
  top: "data"
  top: "label"
  data_param {
    source: "Sandipan1_half_test10_leveldb"
    backend: LMDB
    batch_size: 130
  }
  transform_param {
    crop_size: 227
    mean_file: "imagenet_mean.binaryproto"
    mirror: false
  }
  include: { phase: TEST }

I save the weights after 10000 iterations. The deploy.prototxt file I am using to test is exactly this one: https://github.com/BVLC/caffe/blob/master/models/bvlc_reference_caffenet/deploy.prototxt.

If you see the last layer in the deploy.prototxt file, it is named as 'prob' which I think stands for the probability for the test image being a sample of each class trained the model is trained with. For my data, since I trained with 13 drug classes, it can have class labels from 0-12. So basically it outputs 1 label. out['prob'].argmax() should output the index(label) which has the highest probability value for the test image. But the labels are not all right for some reason when testing them individually, although most of them give accuracy=1 when testing them with Caffe directly (with the 'caffe test...' option), one by one (with batch size=1 for testing).

I don't understand what I am doing wrong, please let me know if you require any other information.

**Comment by shelhamer** [comment]
Closing in favor of the [caffe-users post](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/caffe-users/B3AYQS9el3Y/FBjycEZ9QlUJ) since this is a usage question.
